wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -1088,6 +1097,10 @@ private void shutdown() {
     }
 
     private void createWallet(NetworkParameters params, File walletFile) throws IOException {
+        KeyChainGroupStructure keychainStructure = (structure == Structure.BIP32) ?
```

Comment:
I wonder if this logic shouldn't be moved more global, given that we need the structure for more than just creating wallets?

Reply:
Probably.

Reply:
Probably.